### PR TITLE
Nutrition and hydration adjustments.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Adds procs and vars for handling nutrition and hydration at /mob/living.
- Opts out of this for simple mobs and silicons for the time being.

## Why and what will this PR improve
Allows for unifying a bunch of shared logic.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

File: mods/content/xenobiology/slime/life.dm [hunk 1]
```diff
@@ -80,9 +80,15 @@
...
+/mob/living/slime/get_hunger_factor()
+	return (0.1 + 0.05 * is_adult)
+
+/mob/living/slime/get_thirst_factor()
+	return 0
+
 /mob/living/slime/handle_nutrition_and_hydration()
 
-	adjust_nutrition(-(0.1 + 0.05 * is_adult))
+	..()
```
Comment (out-of-phaze) [inline mods/content/xenobiology/slime/life.dm]: Why was this moved prior to digesting mobs in contents?

File: mods/content/xenobiology/slime/_slime.dm [hunk 2]
```diff
@@ -292,6 +292,16 @@
 /mob/living/slime/check_has_mouth()
 	return 0
 
+/mob/living/slime/set_nutrition(amt)
+	..()
+	if(get_hydration() < amt)
+		set_hydration(amt)
+
+/mob/living/slime/set_hydration(amt)
```
Comment (out-of-phaze) [inline mods/content/xenobiology/slime/_slime.dm]: Why not just set the hydration rate to zero and opt out of hydration altogether? Doing `nutrition = max(nutrition, hydration); hydration = max(hydration, nutrition)` is a bit confusing and I'm not sure what it's meant to solve.

**Review by out-of-phaze — COMMENTED** [review]
A few clarifying questions, then I'll re-review.

File: mods/content/xenobiology/slime/_slime.dm (hunk 2 above)
Comment (MistakeNot4892) [reply mods/content/xenobiology/slime/_slime.dm]: It chains the values to each other as I don't think slimes have a way to get hydration (since water kills them) but the bodytemp code in the other PR will subtract hydration to stabilize bodytype. Chaining them to each other seems like a better solution than always returning max

File: mods/content/xenobiology/slime/life.dm (hunk 1 above)
Comment (MistakeNot4892) [reply mods/content/xenobiology/slime/life.dm]: Didn't seem important, I'll move it beneath.

File: mods/content/xenobiology/slime/_slime.dm (hunk 2 above)
Comment (out-of-phaze) [reply mods/content/xenobiology/slime/_slime.dm]: Is this chaining, though? I feel like this could lead to weird edge cases given the conditional.

Comment (MistakeNot4892) [reply mods/content/xenobiology/slime/_slime.dm]: Replaced with just returning `get_nutrition()` for `get_hydration()`.

**Review by out-of-phaze — APPROVED** [review]
